include/RevProc.h

```diff
@@ -170,29 +170,37 @@ class RevProc{
   void SetCoProc(RevCoProc* coproc);
 
 //--------------- External Interface for use with Co-Processor -------------------------
+  ///< RevProc: Allow a co-processor to query the bits in scoreboard. Note the RevProcPassKey may only
+  ///  be created by a RevCoProc (or a class derived from RevCoProc) so this function may not be called from even within
+  ///  RevProc
+  bool ExternalDepCheck(RevProcPasskey<RevCoProc>, uint16_t HartID, uint16_t RegNum, bool IsFloat){
+    const RevRegClass regClass = IsFloat ? RevRegClass::RegFLOAT : RevRegClass::RegGPR;
+    return DependencyCheck(HartID, RegNum, regClass);
+  }
+
   ///< RevProc: Allow a co-processor to manipulate the scoreboard by setting a bit. Note the RevProcPassKey may only
-  ///  be created by a RevCoProc (or a class derived from RevCoProc) so this funciton may not be called from even within
+  ///  be created by a RevCoProc (or a class derived from RevCoProc) so this function may not be called from even within
   ///  RevProc
   void ExternalDepSet(RevProcPasskey<RevCoProc>, uint16_t HartID, uint16_t RegNum, bool isFloat, bool value = true){
```

Comment:
HartID should be `unsigned` as that is what the SST parameter `numHarts` is parsed as